dynamo/preprocessing/preprocessor_utils.py

```diff
@@ -806,13 +804,16 @@ def normalize_cell_expr_by_size_factors(
             scale_to=scale_to,
         )
 
+    layers = pd.Series(layers_to_sz).str.split("_Size_Factor", expand=True).iloc[:, 0].tolist()
+    layers[np.where(np.array(layers) == "Size_Factor")[0][0]] = "X"
+
     excluded_layers = []
```

Comment:
why do you have this change here? what I wrote before will try to get the layer names from layer_sz_column_names and then append X to the layers. Then we call cal_sz_factor function to get the size factor for each layer

Reply:
I understand your intention, but I believe this change achieves the same result as before and is more concise. Furthermore, by removing the two if statement lines that always return True, we can simplify the code and potentially improve its performance..

```
layer_sz_column_names = [i + "_Size_Factor" for i in set(layers).difference("X")]
    layer_sz_column_names.extend(["Size_Factor"])
    # layers_to_sz = list(set(layer_sz_column_names).difference(adata.obs.keys()))
    layers_to_sz = list(set(layer_sz_column_names))
    if len(layers_to_sz) > 0:              ## Always True!
        layers = pd.Series(layers_to_sz).str.split("_Size_Factor", expand=True).iloc[:, 0].tolist()
        if "Size_Factor" in layers:       ## Always True!
            layers[np.where(np.array(layers) == "Size_Factor")[0][0]] = "X"
        calc_sz_factor(
            adata,
            layers=layers,
            locfunc=np.nanmean,
            round_exprs=True,
            method=sz_method,
            scale_to=scale_to,
        )
```